File: Project Euler/3.rb
#The prime factors of 13195 are 5, 7, 13 and 29.
#What is the largest prime factor of the number 600851475143 ?

def is_prime?(num)
  2.upto(num - 1) { |n| return false if num % n == 0 }
  return true
end

def largest_prime_factor(number) 
  number = number / 2
  
  loop do
    return number if is_prime?(number)
    number -= 1
  end
end

puts largest_prime_factor(600851475143)

File: Project Euler/3.js
//The prime factors of 13195 are 5, 7, 13 and 29.
//What is the largest prime factor of the number 600851475143 ?

function largestPrimeFactor(n) {
  for (let i = Math.floor(n / 2); i > 3; i--) {
    if (isPrime(i) && n % i === 0) {
      return i;
    }
  }
}

function isPrime(n) {
  for (let i = 3; i < n - 1; i += 2) {
    if (n % i === 0) {
       return false;
    }
  }

  return true;
}
//console.log(isPrime(2639));
console.log(largestPrimeFactor(600851475143));
//console.log(largestPrimeFactor(13195));
